app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,20 @@ def show
     @post = Post.find(params[:id].to_i)
     @post_comments = Comment.where(post: @post)
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @new_post = params.require(:post).permit(:title, :text)
+    @post = Post.new(author: current_user, title: @new_post[:title], text: @new_post[:text], comments_counter: 0,
+                     likes_counter: 0)
+    if @post.save
+      flash[:success] = 'Post created succefully!'
+      redirect_to users_path
+    else
+      flash.now[:error] = "The post couldn't be created!"
+    end
+  end
```

Comment:
- [ ] [OPTIONAL] I would like to recommend you add the params method below :arrow_down: you can check the following [link](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides) to know more about it :+1:

Reply:
Thank you for your feedback, due to the deadline I am skipping the optional tasks but I will surely act upon them on coming Sunday.